File: DSA/Leetcode-3713Ex.js
class Leetcode3713Ex {
    longestBalanced(s) {
        let max = 0;
        for (let i = 0; i < s.length; i++) {
            const a = new Array(26).fill(0);
            for (let j = i; j < s.length; j++) {
                a[s.charCodeAt(j) - 'a'.charCodeAt(0)]++;
                if (this.find(a)) {
                    max = Math.max(max, j - i + 1);
                }
            }
        }
        return max;
    }

    find(a) {
        let c = 0;
        for (let i of a) {
            if (i > 0) {
                if (c === 0) {
                    c = i;
                } else if (i !== c) {
                    return false;
                }
            }
        }
        return true;
    }
}

File: DSA/Leetcode-3713Ex.java
class Leetcode3713Ex {
    public int longestBalanced(String s) {
        int max = 0;
        for (int i = 0; i < s.length(); i++) {
            int[] a = new int[26];
            for (int j = i; j < s.length(); j++) {
                a[s.charAt(j) - 'a']++;
                if (find(a)) {
                    max = Math.max(max, j - i + 1);
                }
            }
        }
        return max;
    }

    private boolean find(int[] a) {
        int c = 0;
        for (int i : a) {
            if (i > 0) {
                if (c == 0) {
                    c = i;
                } else if (i != c) {
                    return false;
                }
            }
        }
        return true;
    }
}